libsolidity/parsing/Parser.cpp

```diff
@@ -1029,7 +1039,7 @@ ASTPointer<Statement> Parser::parseStatement()
 		statement = parseSimpleStatement(docString);
```

Comment:
I think `parseSimpleStatement` would also be a good candidate for a try-catch error recovery. At the end of that, we don't even have to do anything because we call `expectTokenOrConsumeUntil` below.

Reply:
I don't understand. All error tokens which is what we are simulating (in the Steve C Johnson sense) need a token to syncronize to so as to at least try to get things straight. 

How about if I add the try catch to parseStatement which is one level up?  That code is already modified.

One other potential place to catch errors is in the docstring.